### PR TITLE
My Jetpack: expose pricing product data

### PR DESCRIPTION
**Pull request by retrofox**
This PR exposes the pricing data for each product, adding the `get_pricing_for_ui()` static method to the Product class. We can define the way that we are going to get and propagate these data. Hardcoding vs pulling from wpcom.
The data provided should be enough to show the following price section of the Product card:

Screen Shot 2022-02-02 at 6 18 01 PM

Fixes #

#### Changes proposed in this Pull Request:

* My Jetpack: expose pricing product data

#### Testing instructions:

* Go to My Jetpack page
* Check the Product items taking a look at the items:
```
myJetpackInitialState.products.items
```

image

### CODE REVIEW

File: projects/packages/my-jetpack/src/products/class-product.php [hunk 1]
```diff
@@ -92,6 +92,7 @@ public static function get_info() {
 			'long_description'         => static::get_long_description(),
 			'features'                 => static::get_features(),
 			'status'                   => static::get_status(),
+			'pricing_for_ui'           => static::get_pricing_for_ui(),
```
Comment (renatoagds) [inline projects/packages/my-jetpack/src/products/class-product.php:95]: Why add `for_ui` suffix? I think just `pricing` makes sense, not?

Comment (renatoagds) [reply projects/packages/my-jetpack/src/products/class-product.php:95]: Reasons explained here: p1643837371383059-slack-C02TQF5VAJD

**Review by renatoagds — APPROVED** [review]

**Comment by sdixon194** [comment]
Looks like some PHP errors might need to be cleaned up before it can be merged.

File: projects/packages/my-jetpack/src/products/class-product.php (hunk 1 above)
Comment (retrofox) [reply projects/packages/my-jetpack/src/products/class-product.php:95]: sorry, thanks for the link

**Review by renatoagds — CHANGES_REQUESTED** [review]
Change review, based on @sdixon194 comment

**Review by sdixon194 — APPROVED** [review]
Things look to be passing now 👍

**Comment by retrofox** [comment]
> Changing review, based on @sdixon194 comment

I don't get why you are requesting changes here, @renatoagds 

The PR was blocked because of some failing tests, already addressed here https://github.com/Automattic/jetpack/pull/22630/commits/5f761109fee5db9ac6a247cdbb672c5b10bb4f4c

**Comment by renatoagds** [comment]
@retrofox I requested changes at the same time you did a new push, did based on the problems before you fix them.

**Review by renatoagds — APPROVED** [review]

**Pull request merged**